src/main/java/de/plushnikov/intellij/plugin/provider/LombokAugmentProvider.java

```diff
@@ -46,26 +51,44 @@ public LombokAugmentProvider() {
     modifierProcessors = Arrays.asList(getModifierProcessors());
   }
 
-  @Nullable
+  /**
+   * Support method required by patcher project and {@link ModifierVisibilityClassFileTransformer}.
+   * Provides a simple way to inject modifiers into older versions of IntelliJ. Return of the null value is dictated by legacy IntelliJ API.
+   *
+   * @param modifierList PsiModifierList that is being queried
+   * @param name         String name of the PsiModifier
+   * @return {@code Boolean.TRUE} if modifier exists (explicitly set by modifier transformers of the plugin), {@code null} otherwise.
+   */
+  public Boolean hasModifierProperty(@NotNull PsiModifierList modifierList, @NotNull final String name) {
+
+    final Set<String> modifiers = this.transformModifiers(modifierList, Collections.<String>emptySet());
+
+    if (modifiers.contains(name)) {
+      return Boolean.TRUE;
+    }
+
+    return null;
+  }
+
+  @NotNull
   //@Override //May cause issues with older versions of IDEA SDK that are currently supported
-  protected Boolean hasModifierProperty(@NotNull PsiModifierList modifierList, @NotNull String name) {
+  protected Set<String> transformModifiers(@NotNull PsiModifierList modifierList, @NotNull final Set<String> modifiers) {
+
     if (DumbService.isDumb(modifierList.getProject())) {
```

Comment:
We don't need it here any more, because IntelliJ already checked for it. But maybe inside of the backported hasModifiersProperty it could be still useful.